pallets/teebag/src/lib.rs

```diff
@@ -0,0 +1,833 @@
+			ensure!(worker_type.is_sidechain(), Error::<T>::ScheduledEnclaveNotApplicable,);
+			ScheduledEnclave::<T>::insert((worker_type, sidechain_block_number), mrenclave);
+			Self::deposit_event(Event::ScheduledEnclaveSet {
+				worker_type,
+				sidechain_block_number,
+				mrenclave,
+			});
+			Ok(().into())
+		}
+
+		#[pallet::call_index(8)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal, Pays::No))]
+		pub fn remove_scheduled_enclave(
+			origin: OriginFor<T>,
+			worker_type: WorkerType,
+			sidechain_block_number: SidechainBlockNumber,
+		) -> DispatchResultWithPostInfo {
+			Self::ensure_admin_or_root(origin)?;
+			ensure!(worker_type.is_sidechain(), Error::<T>::ScheduledEnclaveNotApplicable,);
+			ensure!(
+				ScheduledEnclave::<T>::contains_key((worker_type, sidechain_block_number)),
+				Error::<T>::ScheduledEnclaveNotExist
+			);
+			ScheduledEnclave::<T>::remove((worker_type, sidechain_block_number));
+			Self::deposit_event(Event::ScheduledEnclaveRemoved {
+				worker_type,
+				sidechain_block_number,
+			});
+			Ok(().into())
+		}
+
+		#[pallet::call_index(9)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal, Pays::Yes))]
+		pub fn register_enclave(
+			origin: OriginFor<T>,
+			worker_type: WorkerType,
+			attestation: Vec<u8>,
+			worker_url: Vec<u8>,
+			shielding_pubkey: Option<Vec<u8>>,
+			vc_pubkey: Option<Vec<u8>>,
+			attestation_type: AttestationType,
+		) -> DispatchResultWithPostInfo {
+			let sender = ensure_signed(origin)?;
+
+			ensure!(worker_url.len() <= MAX_URL_LEN, Error::<T>::EnclaveUrlTooLong);
+
+			let mut enclave = Enclave::new(
+				worker_type,
+				worker_url,
+				shielding_pubkey,
+				vc_pubkey,
+				attestation_type,
+			);
+			match attestation_type {
+				AttestationType::Ignore => {
+					ensure!(
+						Self::mode() == OperationalMode::Development,
+						Error::<T>::InvalidAttestationType
+					);
+					enclave.mrenclave =
+						<MrEnclave>::decode(&mut attestation.as_slice()).unwrap_or_default();
+					enclave.last_seen_timestamp = Self::now().saturated_into();
+					enclave.sgx_build_mode = SgxBuildMode::default();
+				},
+				AttestationType::Ias => {
+					let report = Self::verify_ias(&sender, attestation)?;
+					enclave.mrenclave = report.mr_enclave;
+					enclave.last_seen_timestamp = report.timestamp;
+					enclave.sgx_build_mode = report.build_mode;
+				},
+				AttestationType::Dcap(_) => {
+					let report = Self::verify_dcap(&sender, attestation)?;
+					enclave.mrenclave = report.mr_enclave;
+					enclave.last_seen_timestamp = report.timestamp;
+					enclave.sgx_build_mode = report.build_mode;
+				},
+			};
+
+			match Self::mode() {
+				OperationalMode::Production | OperationalMode::Maintenance => {
+					if !Self::allow_sgx_debug_mode() &&
+						enclave.sgx_build_mode == SgxBuildMode::Debug
+					{
+						return Err(Error::<T>::InvalidSgxMode.into())
+					}
+					// TODO: shall we take the sidechain number into consideration?
+					ensure!(
```

Comment:
I would keep it as it is if we really need this check.

Reply:
Yea I think it will have similar effect.

If the sidechain block number doesn't match, it won't produce blocks.
It's just if we should allow it to register at all

I removed the comment

Reply:
I think it's still not removed 😂

Reply:
Yes I didn't push it - now it's there :)